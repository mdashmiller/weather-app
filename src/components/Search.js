import styled, { css } from "styled-components"

const Search = styled.input`
	background: #eee;
	border: 1px solid #c3c3c3;
	border-radius: 10px;
	margin: 200px auto 20px;
	width: 200px;
	height: 34px;
	padding-left: 8px;
	font-family: 'Font Awesome 5 Free', 'Varela Round', Arial, sans-serif;
	font-weight: 900;
	font-size: 18px;

	&::placeholder {
		color: #6a6a6a;
	}

	${props => props.short && css`
	    width: 114px;
	    padding-left: 5px;
	    margin: 0;
	  `}

	${props => props.landingPage && css`
		display: block;
	    width: 172px;
	   	margin: 26px auto;
		text-align: center;
		padding: 0;
		cursor: pointer;  
	  `}

	${props => props.userLocation && css`
		border: 3px solid #f46403;
			
			&:hover {
				background: #f46403;
				
				&::placeholder {
					color: #eee;
				}
			}	
	  `}

	${props => props.userZip && css`
		border: 3px solid #6a6a6a;

			&:hover {
				background: #6a6a6a;
				
				&::placeholder {
					color: #eee;
				}
			} 
	  `}
`

export default Search
