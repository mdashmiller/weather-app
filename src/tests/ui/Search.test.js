import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Search from '../../components/ui/Search'

describe('<Search /> rendering', () => {

	it('renders without crashing', () => {
	  	const div = document.createElement('div')
	  	ReactDOM.render(<Search />, div)
	  	ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(<Search />)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

})
