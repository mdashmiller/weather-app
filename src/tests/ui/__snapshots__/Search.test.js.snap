// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> rendering has a valid snapshot 1`] = `
<input
  className="sc-bdVaJa gIfCVv"
/>
`;
